app/src/main/kotlin/co/touchlab/droidconandroid/ui/DrawerAdapter.kt

```diff
@@ -8,139 +9,100 @@ import android.view.LayoutInflater
+                return HeaderViewHolder(view)
+            }
+            else -> {
+                val view = LayoutInflater.from(context).inflate(R.layout.item_drawer_divider, parent, false)
+                return DividerViewHolder(view)
             }
-
         }
-        else if (getItemViewType(position) == VIEW_TYPE_NAVIGATION)
-        {
-            val navItem = dataSet[position] as NavigationItem
-            val navHolder = holder as NavigationViewHolder
-            navHolder.itemView.title.setText(navItem.titleRes)
-            val drawable = ResourcesCompat.getDrawable(context, navItem.iconRes)
+    }
 
-            val selected = selectedPos == position
-            navHolder.itemView.isSelected = selected
-            if (selected)
-            {
-                drawable.colorFilter = PorterDuffColorFilter(ContextCompat.getColor(context,
-                        R.color.nav_icon_highlight), PorterDuff.Mode.SRC_IN)
-                navHolder.itemView.highlight.visibility = View.VISIBLE
-            }
-            else
-            {
-                drawable.colorFilter = PorterDuffColorFilter(ContextCompat.getColor(context,
-                        R.color.drawer_icons), PorterDuff.Mode.SRC_IN)
-                navHolder.itemView.highlight.visibility = View.GONE
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder?, position: Int) {
+        when (getItemViewType(position)) {
+            VIEW_TYPE_HEADER -> {
+                val headerHolder = holder as HeaderViewHolder
+                headerHolder.itemView.setOnClickListener {
+                    if (selectedPos != position)
+                        drawerClickListener.onHeaderItemClick()
+                }
             }
-            navHolder.itemView.icon.setImageDrawable(drawable)
-            navHolder.itemView.action_icon.visibility = if (navItem.showAction) View.VISIBLE else View.GONE
+            VIEW_TYPE_NAVIGATION -> {
+                val navItem = dataSet[position] as NavigationItem
+                val view = (holder as NavigationViewHolder).itemView
+                view.title.setText(navItem.titleRes)
+                val drawable = ContextCompat.getDrawable(context, navItem.iconRes)
+
+                view.isSelected = selectedPos == position
+                if (view.isSelected) {
+                    drawable.colorFilter = PorterDuffColorFilter(ContextCompat.getColor(context,
+                            R.color.nav_icon_highlight), PorterDuff.Mode.SRC_IN)
+                    view.highlight.visibility = View.VISIBLE
+                } else {
+                    drawable.colorFilter = PorterDuffColorFilter(ContextCompat.getColor(context,
+                            R.color.drawer_icons), PorterDuff.Mode.SRC_IN)
+                    view.highlight.visibility = View.GONE
+                }
+                view.icon.setImageDrawable(drawable)
+                view.action_icon.setViewVisibility(navItem.showAction)
```

Comment:
I just tested it - action_icon is only visible on "Chat on slack" option, but making it invisible vs gone has no visual effect because icon is in line with text.